### PR TITLE
Case 22322: Oven handles TextureBakers with same URL but different usage + fix domain baking

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/22322/Oven-DomainBaker-needs-to-take-into-account-TextureUsage-when-baking-textures

Test plan:
- Create a Zone with the same skybox URL and ambient map URL.  Export the zone using Create.
- Bake the resulting JSON using the oven from this PR.
- Unzip the generated models.json.gz.  The ambient and skybox urls should be different.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18882/HighFidelity-Beta-PR15439-7245d1a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18882/HighFidelity-Beta-Interface-PR15439-7245d1a.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18882/HighFidelity-Beta-PR15439-7245d1a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18882/HighFidelity-Quest-Beta-PR15439-7245d1a.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18882/HighFidelity-Beta-PR15439-7245d1a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18882/HighFidelity-Beta-Interface-PR15439-7245d1a.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18940/HighFidelity-Beta-PR15439-3b74638.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18940/HighFidelity-Quest-Beta-PR15439-3b74638.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18940/HighFidelity-Beta-PR15439-3b74638.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18940/HighFidelity-Beta-Interface-PR15439-3b74638.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18940/HighFidelity-Beta-PR15439-3b74638.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18940/HighFidelity-Beta-Interface-PR15439-3b74638.exe

File: tools/oven/src/DomainBaker.cpp [hunk 1]
```diff
@@ -220,7 +222,8 @@ void DomainBaker::addTextureBaker(const QString& property, const QString& url, i
 
         // add this QJsonValueRef to our multi hash so that it can re-write the texture URL
         // to the baked version once the baker is complete
-        _entitiesNeedingRewrite.insert(textureURL, { property, jsonRef });
+        // it doesn't really matter what this key is as long as it's consistent
+        _entitiesNeedingRewrite.insert(textureURL.toDisplayString() + type, { property, jsonRef });
```
Comment (sabrina-shanman) [inline tools/oven/src/DomainBaker.cpp]: Just in case, can we throw in an invalid URL character in here to make sure there's no name collisions?

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18985/HighFidelity-Beta-PR15439-a66c515.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18985/HighFidelity-Quest-Beta-PR15439-a66c515.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18985/HighFidelity-Beta-PR15439-a66c515.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18985/HighFidelity-Beta-Interface-PR15439-a66c515.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18985/HighFidelity-Beta-PR15439-a66c515.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18985/HighFidelity-Beta-Interface-PR15439-a66c515.exe

**Comment by sabrina-shanman** [comment]
URLs are still not being replaced in models.json.gz

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19007/HighFidelity-Beta-PR15439-cde8498.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19007/HighFidelity-Beta-Interface-PR15439-cde8498.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19007/HighFidelity-Beta-PR15439-cde8498.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19007/HighFidelity-Beta-Interface-PR15439-cde8498.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19007/HighFidelity-Beta-PR15439-cde8498.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19007/HighFidelity-Quest-Beta-PR15439-cde8498.apk)

**Pull request merged**